Source/WebCore/Modules/WebGPU/GPUDevice.cpp

```diff
@@ -111,6 +112,23 @@ void GPUDevice::destroy()
     m_backing->destroy();
 }
 
+GPUDevice::LostPromise& GPUDevice::lost()
+{
+    if (m_waitingForDeviceLostPromise)
+        return m_lostPromise;
+
+    m_waitingForDeviceLostPromise = true;
+    m_backing->resolveDeviceLostPromise([weakThis = WeakPtr { *this }] (PAL::WebGPU::DeviceLostReason reason) {
```

Comment:
I used `WeakPtr` here to avoid a retain cycle where GPUDevice's destructor would never execute